src/cosmic_ray/ast.py

```diff
@@ -43,6 +47,40 @@ def get_ast(module_path, python_version):
     return parso.parse(source, version=python_version)
 
 
+def get_comment_on_node_line(node) -> str or None:
+    """
+    From a parso node, get the comment on the node line
+    and return the comment
+    """
+
+    while not isinstance(node, Leaf):
+        node = node.children[0]
+
+    # Now we are looking for any non empty prefix before next '\n'
+    while node is not None:
+        node = node.get_next_leaf()
+        if node:
+            # don't strip '\n'
+            prefix = node.prefix.strip(" \t")
+            if prefix:
+                return prefix
+
+        if isinstance(node, parso.python.tree.Newline):
+            return node.prefix
+
+
+def get_node_pragma_categories(node) -> None or Dict[str, None or List[str]]:
```

Comment:
"pragma" is not a core CR concept. Any handling of pragmas needs to take place in the isolation of the appropriate interceptor.